react/react-training/src/app/screens/Game/components/Board/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import Square from './components/Square/';
+import styles from './styles.scss';
+
+class Board extends Component {
+  handleClick = i => {
+    const { onClick } = this.props;
+    onClick(i);
+  };
+
+  renderSquare = i => {
+    const { squares } = this.props;
+    return <Square value={squares[i]} squareNumber={i} onClick={this.handleClick} />;
+  };
+
+  render() {
+    return (
+      <div className={styles.board}>
+        <div className={styles.status}>{status}</div>
+        <div className={styles.row}>
+          {this.renderSquare(0)}
+          {this.renderSquare(1)}
+          {this.renderSquare(2)}
+        </div>
+        <div className={styles.row}>
+          {this.renderSquare(3)}
+          {this.renderSquare(4)}
+          {this.renderSquare(5)}
+        </div>
+        <div className={styles.row}>
+          {this.renderSquare(6)}
+          {this.renderSquare(7)}
+          {this.renderSquare(8)}
+        </div>
+      </div>
+    );
+  }
+}
+
+Board.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  squares: PropTypes.arrayOf.isRequired
```

Comment:
U always need to declare what type is in the array. `PropTypes.arrayOf(PropTypes.number.isRequired).isRequired`